### PR TITLE
Implemented PUT method

### PR DESCRIPTION
**Pull request by rob-deans**
Also renamed the old "PUT" method to the correct name for the implementation which is "PATCH"

Implements: #11

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -3,6 +3,6 @@ import PackageDescription
 let package = Package(
     name: "GeneratedSwiftServer",
     dependencies: [
-        .Package(url: "https://github.com/IBM-Swift/Kitura.git", majorVersion: 1, minor: 1),
+        .Package(url: "https://github.com/IBM-Swift/Kitura.git", majorVersion: 1),
```
Comment (tunniclm) [inline Package.swift]: I don't think we want to change this.

File: Sources/MemoryStore.swift [hunk 2]
```diff
@@ -52,6 +52,25 @@ public class MemoryStore: Store {
...
 
+    private func changeID(from memoryStoreID: MemoryStoreID, to newID: Any?, type: Model.Type, callback: @escaping EntityCallback) throws -> MemoryStoreID {
+        if let newID = try newID.map({ try MemoryStoreID($0) }) {
+            if newID.value != memoryStoreID.value {
+                // Changing the id of this model
+                if let (_, _) = findOne_(type: type, id: newID) {
+                    // NOTE(tunniclm): The new id must not clash with existing model
+                    callback(nil, .idConflict(newID))
+                    return memoryStoreID
+                }
```
Comment (tunniclm) [inline Sources/MemoryStore.swift:63]: This doesn't look like it is defining a good error path.

**Review by tunniclm — COMMENTED** [review]
This has some breaking changes, so we need to bump the minor version when it gets merged.

File: Package.swift (hunk 1 above)
Comment (rob-deans) [reply Package.swift]: Do we want to go to 1.2?

**Comment by tunniclm** [comment]
Gave this a little test and lgtm.
I'm going to merge, but we can't update the generator to the new version until CloudantStore is updated (https://github.com/IBM-Swift/GeneratedSwiftServer-CloudantStore/issues/4).

**Pull request merged**